tests/test_pagination.py

```diff
@@ -204,6 +204,19 @@ def test_to_dict(self):
             p.to_dict()
         )
 
+    def test_data_deprecation(self):
+        p = self.paginator(get_response, total=2)
+        if p._is_caching:
```

Comment:
When will `p._is_caching` be false? There will be missing coverage when it is.

Reply:
I was thinking about making a skipif for various cases in this test, but left it for now. `_is_caching` for the identical test sequence `TestNoCache`, for which a small amount of functionality varies (namely, without the cache you can't repeatedly list an exhausted iterator).

Reply:
Could you please add a comment to the code?